tools/devtool

```diff
@@ -275,8 +275,40 @@ validate_version() {
     if [ -z "$version" ]; then
         die "Version cannot be empty."
     elif [[ ! "$version" =~ $version_regex ]]; then
-        die "Invalid version number: $version (expected: \$Major.\$Minor.\$Build).".
+        die "Invalid version number: $version (expected: \$Major.\$Minor.\$Build)."
     fi
+
+}
+
+# Compose the text for a new release tag using the information in the changelog,
+# between the two specified releases.
+# The following transformations are applied:
+# * `-` is replaced with `*` for unnumbered lists.
+# * section headers (`###`) are removed.
+#
+# Args:
+#   $1  previous version.
+#   $2  new version.
+#
+compose_tag_text() {
+    declare prev_ver="$1"
+    declare curr_ver="$2"
+    declare changelog="$FC_ROOT_DIR/CHANGELOG.md"
+
+    validate_version "$prev_ver"
+    validate_version "$curr_ver"
+
+    # Patterns for the sections in the changelog corresponding to the versions.
+    pat_prev="## \[$prev_ver\]"
+    pat_curr="## \[$curr_ver\]"
```

Comment:
I used `\s` instead of `[[:space:]]` because the latter wanted the `-E` switch for `sed`, which isn't available on my setup.

Reply:
I hope that I never have to change that line of code.

Reply:
I am tempted to defer this unholy match to an inline python with a nice regex that won't make future developers wince.

Reply:
Okay, okay, there's no need for that. Please put the python down...

If we must rely on some dependency, we should use the container (i.e. run whatever we need to run inside the container), since that's what we use to ensure a sane environment (all dependencies, all proper versions, etc).

For this particular case, though, bash, ugly as it is, will do. Most of the ugliness comes from having to escape the regex, which would have to happen with python as well (it'll still be sent from bash via command line).